### PR TITLE
[FSDE-30354] Implement tooltip with directions for icon component

### PR DESCRIPTION
**Pull request by soujanreddy**

### CODE REVIEW

File: src/components/Icon/Icon.js [hunk 1]
```diff
@@ -8,26 +9,44 @@ import './Icon.css';
...
-    size={size}
-    onClick={click}
-    onMouseEnter={mouseOver}
-    onMouseLeave={mouseLeave}
-    onMouseDown={mouseDown}
-    onMouseUp={mouseUp}>
+const Icon = ({tooltipMessage, tooltipPosition, click, mouseDown, mouseUp, mouseLeave, mouseOver, icon, color, size, uxpinRef}) => {
+  const uuid = uuid4();
+
+  if (tooltipMessage && tooltipPosition !== 'default') {
```
Comment (josecarlosgonzalezv) [inline src/components/Icon/Icon.js:15]:
```suggestion
  if (tooltipMessage) {
```

File: src/components/Icon/Icon.js [hunk 2]
```diff
@@ -8,26 +9,44 @@ import './Icon.css';
...
-);
+  )
+};
 
 Icon.propTypes = {
   size: PropTypes.oneOf(['xs', 'sm', 'sm--2', 'sm--3', 'md', 'lg', 'xl', 'xxl']),
   icon: PropTypes.string,
   color: PropTypes.oneOf(['primary', 'secondary', 'dark', 'light', 'info', 'grey', 'muted', 'success', 'warning', 'danger', 'navy', 'orange']),
+  tooltipMessage: PropTypes.string,
+  tooltipPosition: PropTypes.oneOf(['default', 'top', 'right', 'bottom', 'left']),
```
Comment (josecarlosgonzalezv) [inline src/components/Icon/Icon.js:49]:
```suggestion
  tooltipPosition: PropTypes.oneOf(['top', 'right', 'bottom', 'left']),
```

File: src/components/Icon/Icon.js [hunk 3]
```diff
@@ -40,6 +59,8 @@ Icon.defaultProps = {
   size: 'sm',
   color: 'primary',
   icon: 'atom',
+  tooltipMessage: 'Tooltip message',
+  tooltipPosition: 'default',
```
Comment (josecarlosgonzalezv) [inline src/components/Icon/Icon.js:63]:
```suggestion
  tooltipPosition: 'top',
```

**Review by josecarlosgonzalezv — CHANGES_REQUESTED** [review]

File: src/components/Icon/Icon.js (hunk 2 above)
Comment (soujanreddy) [reply src/components/Icon/Icon.js:49]: @josecarlosgonzalezv As per the discussion with Tom, he did not wanted tooltip text to be shown on top position by default, it must be empty. Only upon selecting a particular tooltip position from dropdown should reveal the tooltip on the icon component. Hence, i have implemented the default property.

Comment (soujanreddy) [reply src/components/Icon/Icon.js]: @josecarlosgonzalezv As you mentioned, had a discussion with Tom today, changing the default position to top to be consistent with other components.

**Review by jllr — APPROVED** [review]

**Review by josecarlosgonzalezv — APPROVED** [review]

**Pull request merged**